baybe/searchspace/core.py

```diff
@@ -241,6 +241,14 @@ def contains_rdkit(self) -> bool:
             p.encoding is SubstanceEncoding.RDKIT for p in self.discrete.parameters
         )
 
+    @property
```

Comment:
TBH, I not convinced of these properties, at least not of their current form. The two problems I see here:
* We artificially couple our search space to concepts that lie outside its responsibilities, i.e. kernels and definitions like "default kernel" (<-- what is that even supposed to mean???), even if that's done only verbally.
* You are effectively moving the decision how to treat kernel dimensions (which should naturally belong to the surrogate) to the searchspace, which should not have to bother about these things at all. In particular, who says that all surrogates have to treat the task dimension separately? Currently, this is a very special setting ONLY for the GP process – other surrogate might do it differently, and even future versions of the GP (e.g. when introducing multi-fidelity etc.) could deviate.

So if you really think it's worth having these properties, which I don't reject in general, please let's rewrite them such that the terminology and the implied decisions remain completely neutral and decoupled from other classes. For example, instead of talking about kernel and stuff, on can simply stick to terminology that is known to the class (or classes below), such as task parameters. (perhaps: `comp_rep_n_non_task_dimensions` ?)

Reply:
the idea is based on the fact that we currently have one overreaching kernel type for all comp parameter columns, with the exception of the task parameter. I simply call this the default kernel and this property is supposed to compute how many dimensions would be affected (= all dims - task dims) - that this could not be the case in current code is news to me. 

If the future eg introduces parameter-specific kernels they could simply be subtracted here too (your name suggestion then wouldnt make sense tho). In this package, any talk about kernel implies GP, if a surrogate doesnt use the GP kernels, it wouldnt use this property

Its related to GP kernels so not ahving that int he name imo doesnt make sense. If your arg is about coupling then I'd rather remove this property, its not very needed. Let me know

Reply:
Yes, it's about decoupling and perhaps even more about which object is responsible for making the decision how the dimensions are to be divided (see answer in other thread)

Reply:
property removed